File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        for i in range(len(s)):
            if s[i] in ')]}' and len(stack)==0:
                return False
            if s[i] in '([{':
                stack.append(s[i])
            elif s[i] in ')]}':
                if s[i]==')' and stack[-1]=='(':
                    stack.pop(-1)
                elif s[i]==']' and stack[-1]=='[':
                    stack.pop(-1)
                elif s[i]=='}' and stack[-1]=='{':
                    stack.pop(-1)
                else:
                    return False
        if len(stack)==0:
            return True
        return False

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();

        for(char i : s.toCharArray()){
            if(stk.isEmpty() || i=='(' || i=='[' || i=='{')
                stk.push(i);
            else{
                char p = stk.peek();
                if((p=='('&& i==')') || (p=='[' && i==']') || (p=='{' && i=='}'))
                    stk.pop();
                else
                    return false;
            }
        }

        if(stk.isEmpty())
            return true;
        return false;
    }
}